### PR TITLE
Enable Nav2 in andino_rmf

### PR DESCRIPTION
**Pull request by ptientho**
This PR intends to add or modify codes to enable Nav2 features to our andino_rmf package.

Several changes include

- add nav2 flag to use within the **andino_rmf** package following the feature in **andino_gz** that enables Nav2.
- modify **fleet manager** node to include nav2 controller.
- modify and configure **fleet adapter** to extend its functionality using Nav2.
- test launching nav2-enabled fleet with Open-rmf

### CODE REVIEW

File: andino_fleet/config/spawn_robots.yaml
Comment (Voldivh) [inline andino_fleet/config/spawn_robots.yaml:1]: Why are we commenting out the last two andinos?

File: andino_fleet_adapter/config.yaml
Comment (Voldivh) [inline andino_fleet_adapter/config.yaml:1]: Same message about comments on the andino

File: andino_rmf_sim/launch/andino_office.launch.py
Comment (Voldivh) [inline andino_rmf_sim/launch/andino_office.launch.py:1]: If we are not going to use anything that is commented, let's delete it

File: andino_rmf_sim/launch/andino_nav2.launch.py
Comment (Voldivh) [inline andino_rmf_sim/launch/andino_nav2.launch.py:1]: Since this launch file can be used to launch the simulation with either the nav2 controller or a custom one, let's modify the name to be something related to the environment which would be the office world by default. We can delete this file and only modify the other launch file

File: andino_fleet/andino_fleet/fleet_manager_server.py
Comment (Voldivh) [inline andino_fleet/andino_fleet/fleet_manager_server.py]: NIT: Order alphabetically between each import section.

**Review by Voldivh — CHANGES_REQUESTED** [review]
Looks really good! Some comments before we can approve, remember to sign off your commits with the flag `-s`

File: andino_fleet/config/spawn_robots.yaml
Comment (ptientho) [reply andino_fleet/config/spawn_robots.yaml:1]: Because I wanted to test the code with the minimum requirement by launching two robots.

File: andino_rmf_sim/launch/andino_nav2.launch.py
Comment (ptientho) [reply andino_rmf_sim/launch/andino_nav2.launch.py:1]: To enable Nav2 and use it, we first need to set initial robot positions. That is the reason I created this launch file - to set initial positions in Rviz, and then launch rmf in the other launch file. 

I can try setting initial positions automatically in the launch file so that we can use only one file.

**Comment by ptientho** [comment]
In order for nav2 system to run with rmf, first we need to set initial robot positions for all robots. In this situation, we have four robots, therefore, we require four initial positions. 

In this commit, I added a new service to the manager called "**initial_pose_server**", which uses a custom message **InitPose** that looks like
```
string[] robot_names
---
# result whether the poses of all robots are initialized
bool result
```
Here it takes the robot names. Example is the following
```
ros2 service call /initial_pose_server andino_fleet_msg/srv/InitPose "{robot_names: ['andino1', 'andino2', 'andino3', 'andino4']}"
```
This sets initial positions for four robots.

In addition to the initial position feature, I separated high-level launch files into two main files
1) **andino_manager.launch.py** : for everything related to simulation and fleet manager
2) **andino_office_rmf.launch.py**: for everything related to rmf

### Custom Controller
To launch system with custom controllers
```
ros2 launch andino_rmf_sim andino_manager.launch.py
```
In a new terminal, launch
```
ros2 launch andino_rmf_sim andino_office_rmf.launch.py
```

### Nav2
To launch system with nav2 controllers
```
ros2 launch andino_rmf_sim andino_manager.launch.py nav2:=True
```
In a new terminal, set initial positions and run rmf
```
source install/setup.bash
ros2 service call /initial_pose_server andino_fleet_msg/srv/InitPose "{robot_names: ['andino1', 'andino2', 'andino3', 'andino4']}"
ros2 launch andino_rmf_sim andino_office_rmf.launch.py
```

**Review by Voldivh — APPROVED** [review]
LGTM, I'm aware that at the moment we are unable to launch eveything in a single launch file, however, I would like that to happen. We can address it on a follow up PR. Very good job @ptientho !

**Comment by Voldivh** [comment]
@ptientho We need to sign-off the commits in order to be able to merge, you can amend the commit and sign-off the commit by adding the flag `-s` on the `git commit` command. Another solution is to create an empty commit and sign-off that one.

**Pull request merged**
